From d997ad034b6a2e840fefb8224bfd85391cafd71d Mon Sep 17 00:00:00 2001
From: Chris Packham <judge.packham@gmail.com>
Date: Wed, 28 Nov 2018 10:32:00 +1300
Subject: [PATCH] ARM: mvebu: add revision id for Armada-385 B0

Marvell have release a B0 revision of the Armada-385 SoC. This fixes a
hardware errata enabling RGMII to work when the Ethernet voltage is
configured to 3.3V.

Signed-off-by: Chris Packham <judge.packham@gmail.com>
Reviewed-by: Stefan Roese <sr@denx.de>
Signed-off-by: Stefan Roese <sr@denx.de>
---
 arch/arm/mach-mvebu/cpu.c                     | 3 +++
 arch/arm/mach-mvebu/include/mach/soc.h        | 1 +
 arch/arm/mach-mvebu/serdes/a38x/sys_env_lib.h | 1 +
 3 files changed, 5 insertions(+)

diff --git a/arch/arm/mach-mvebu/cpu.c b/arch/arm/mach-mvebu/cpu.c
index 0d2d39878d3..aa1be8ebabc 100644
--- a/arch/arm/mach-mvebu/cpu.c
+++ b/arch/arm/mach-mvebu/cpu.c
@@ -261,6 +261,9 @@ int print_cpuinfo(void)
 		case MV_88F68XX_A0_ID:
 			puts("A0");
 			break;
+		case MV_88F68XX_B0_ID:
+			puts("B0");
+			break;
 		default:
 			printf("?? (%x)", revid);
 			break;
diff --git a/arch/arm/mach-mvebu/include/mach/soc.h b/arch/arm/mach-mvebu/include/mach/soc.h
index 6e2e14efe0b..01577f469b0 100644
--- a/arch/arm/mach-mvebu/include/mach/soc.h
+++ b/arch/arm/mach-mvebu/include/mach/soc.h
@@ -27,6 +27,7 @@
 /* A38x revisions */
 #define MV_88F68XX_Z1_ID	0x0
 #define MV_88F68XX_A0_ID	0x4
+#define MV_88F68XX_B0_ID	0xa
 
 /* TCLK Core Clock definition */
 #ifndef CONFIG_SYS_TCLK
diff --git a/arch/arm/mach-mvebu/serdes/a38x/sys_env_lib.h b/arch/arm/mach-mvebu/serdes/a38x/sys_env_lib.h
index 50b23582665..365332d2b04 100644
--- a/arch/arm/mach-mvebu/serdes/a38x/sys_env_lib.h
+++ b/arch/arm/mach-mvebu/serdes/a38x/sys_env_lib.h
@@ -233,6 +233,7 @@
 /* A38x revisions */
 #define MV_88F68XX_Z1_ID		0x0
 #define MV_88F68XX_A0_ID		0x4
+#define MV_88F68XX_B0_ID		0xa
 /* A39x revisions */
 #define MV_88F69XX_Z1_ID		0x2
 